### PR TITLE
add association into different classes

### PR DESCRIPTION
**Pull request by aradradev**
In this Pull Request, I've:
- Created a class Classroom with the following:

- [x] @label instance variable, should be initialized in the constructor.
- [x] Setter and getter for @label (remember about attr_accessor).

- Created the has-many/belongs-to relationship between Classroom and Student
- Created a class Book
- Created a class Rental
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
 class Student < Person
   attr_reader :classroom
 
-  def initialize(id, classroom, name = 'Unknown', age = nil, parent_permission = true)
+  def initialize(id, _classroom, name = 'Unknown', age = nil, parent_permission = true)
     super(id, name, age, parent_permission)
-    @classroom = classroom
+    @classroom = nil
   end
```
Comment (danmainah) [inline student.rb:16]: - [ ] @aradradev Great work adding your associations however you forgot to add a `classroom` method that takes in parameter classroom and adds it to the @classroom if it's not already there, to fulfill the 'belongs-to' association between student and classroom.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+    rental.book = self
+  end
```
Comment (danmainah) [inline book.rb]: - [ ] @aradradev Great work in your association, however, to achieve `the 'has-many' association` between books and person through rentals, we need to pass the `person, date and book` itself as parameters to create the new rental.

Comment (danmainah) [reply book.rb]: - [ ] @aradradev I didn't see your implementation of the association in your person class, kindly consider adding the association in order to achieve the  the 'has-many' association between books and person through rentals

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @aradradev , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Created the   Classroom and Book class correctly
✔️  Created the belongs-to-association side of Rental and Book correctly.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: person.rb [hunk 3]
```diff
@@ -2,13 +2,14 @@
 
 class Person < Nameable
   attr_accessor :name, :age
```
Comment (franclobo) [inline person.rb]: - [ ] Kindly add `:rentals` to attr_accesor, getters and Setters for @rentals should be present.

File: person.rb [hunk 4]
```diff
@@ -2,13 +2,14 @@
...
-  attr_reader :id
+  attr_reader :id, :rentals
 
   def initialize(id, name = 'Unknown', age = 'nil', parent_permission = true)
     @id = id
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (franclobo) [inline person.rb:13]: - [ ] Kindly add a method for adding rentals. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 👇🏽 

```ruby
def add_rental(book, date)
  Rental.new(date, book, self)
end
```
*[OPTIONAL]* Similar method should be present in book class, just this couple of lines. Well done defining the attributes of the method.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @aradradev, 

# Required Changes ♻️

casito

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Correct Git flow ✔️
- Descriptive PR ✔️
- Set up associations between classes and objects. ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 3 above)
Comment (Abenezer-Tilahun) [reply person.rb]: Fixed.

File: person.rb (hunk 4 above)
Comment (Abenezer-Tilahun) [reply person.rb:13]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @aradradev 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**